packages/rich-text/src/RichTextEditor.tsx

```diff
@@ -0,0 +1,154 @@
+import { EntityProvider } from '@contentful/field-editor-reference';
+import { FieldConnector } from '@contentful/field-editor-shared';
+import * as Contentful from '@contentful/rich-text-types';
+import { Plate, getPlateSelectors, getPlateActions } from '@udecode/plate-core';
+import { css, cx } from 'emotion';
+import deepEquals from 'fast-deep-equal';
+import noop from 'lodash/noop';
+
+import { ContentfulEditorIdProvider, getContentfulEditorId } from './ContentfulEditorProvider';
+import { getPlugins, disableCorePlugins } from './plugins';
+import { SlashCommandsPalette } from './plugins/SlashCommands';
+import { RichTextTrackingActionHandler } from './plugins/Tracking';
+import { documentToEditorValue, normalizeEditorValue, setEditorContent } from './prepareDocument';
+import { styles } from './RichTextEditor.styles';
+import { SdkProvider } from './SdkProvider';
+import Toolbar from './Toolbar';
+import StickyToolbarWrapper from './Toolbar/components/StickyToolbarWrapper';
+import { useOnValueChanged } from './useOnValueChanged';
+
+type ConnectedProps = {
+  sdk: FieldExtensionSDK;
+  onAction?: RichTextTrackingActionHandler;
+  minHeight?: string | number;
+  value?: object;
+  isDisabled?: boolean;
+  onChange?: (doc: Contentful.Document) => unknown;
+  isToolbarHidden?: boolean;
+  actionsDisabled?: boolean;
+};
+
+export const ConnectedRichTextEditor = (props: ConnectedProps) => {
+  const id = getContentfulEditorId(props.sdk);
+  const plugins = React.useMemo(
+    () => getPlugins(props.sdk, props.onAction ?? noop),
+    [props.sdk, props.onAction]
+  );
+
+  const [isFirstRender, setIsFirstRender] = useState(true);
+  const [pendingExternalUpdate, setPendingExternalUpdate] = useState(false);
+
+  const onValueChanged = useOnValueChanged({
+    editorId: id,
+    handler: props.onChange,
+    skip: pendingExternalUpdate || isFirstRender,
+    onSkip: () => setPendingExternalUpdate(false),
+  });
+
+  useEffect(() => {
+    /*
+      This effect is called when the value prop changes. Normally
+      this happens when the value is changed outside of the editor,
+      like in snapshots restoration or remote updates
+      Plate won't update the displayed value on its own, see:
+       - https://github.com/ianstormtaylor/slate/pull/4540
+       - https://github.com/udecode/plate/issues/1169
+
+       The content is forcely set to the new value and it's ensured
+       the change listener isn't invoked
+    */
+    setIsFirstRender(false);
+    const editor = getPlateSelectors(id).editor();
+    if (!editor) {
+      return;
+    }
+    setPendingExternalUpdate(true);
+    setEditorContent(editor, documentToEditorValue(props.value));
+  }, [props.value, id]);
+
+  const classNames = cx(
+    styles.editor,
+    props.minHeight !== undefined ? css({ minHeight: props.minHeight }) : undefined,
+    props.isDisabled ? styles.disabled : styles.enabled,
+    props.isToolbarHidden && styles.hiddenToolbar
```

Comment:
when using `cx` there's a syntax to conditionally add class names such as:
```
cx(styles.editor, {
  [styles.hiddenToolbar]: props.isToolbarHidden,
})
```
when the value of `props.isToolbarHidden` is `true` then it adds the class `hiddenToolbar`, I'd suggest refactoring to the idiomatic syntax.

Reply:
I don't want this branch to diverge from `next`. Let's address these comments as a follow-up when it goes to master